lib/Widen.php

```diff
@@ -167,4 +167,52 @@ public function list_collections( $type = 'global' ) {
 			return $data;
 		}
 	}
+
+	/**
+	 * Returns a specific image size using widen's templated url.
+	 *
+	 * @param string $templated_url The templated url provided by Widen. This is returned in the image response.
+	 * @param int    $width         The target image width.
+	 * @param int    $height        The target image height.
+	 * @param int    $scale         The scale of the image.
+	 */
+	public static function create_url_from_template( $templated_url, $width = null, $height = null, $scale = 1 ): string {
+		$url = Util::sanitize_image_url( $templated_url );
+
+		if ( $width && $height ) {
+			$size = $width . 'x' . $height;
+		} elseif ( $width ) {
+			$size = $width;
+		} else {
+			$size = 'exact';
+		}
+
+		// Perform template string replacements.
+		$url = str_replace( '{size}', $size, $url );
+		$url = str_replace( '{scale}', $scale, $url );
+
+		return $url;
+	}
+
+	/**
+	 * Returns an associate array of image meta for a specific image size.
+	 *
+	 * @param string $templated_url The templated url provided by Widen. This is returned in the image response.
+	 * @param int    $width         The target image width.
+	 * @param int    $height        The target image height.
+	 */
+	public static function get_size_meta( $templated_url, $width = null, $height = null ): array {
+		$file = self::create_url_from_template( $templated_url, $width, $height );
+
+		$image_size = @getimagesize( $file ); // phpcs:ignore WordPress.PHP.NoSilencedErrors.Discouraged
```

Comment:
Silencing errors is typically frowned upon however when WordPress core uses PHP's [getimagesize](https://www.php.net/manual/en/function.getimagesize.php), it is always silenced. I think it has to do with it being used for other file types and not just images, so I am just being consistent here with the larger system that is WordPress.